### PR TITLE
dont need blockservice workers anymore

### PR DESCRIPTION
**Pull request by whyrusleeping**
The worker in blockservice was useless. We buffer outgoing provides in bitswap anyways, removing the workers here will make things a bit simpler.

TODO: 
- [x] delete workers code
- [x] figure out where context should come from

License: MIT
Signed-off-by: Jeromy <jeromyj@gmail.com>

### CODE REVIEW

**Comment by jbenet** [comment]
Agree there's duplicated logic here, but I'm not sure this is right. Would like to see some benchmarks for this, when adding huge things in a networked environment.

- https://github.com/ipfs/go-ipfs/blob/feat/rm-worker/blockservice/worker/worker.go#L82-L119 -- this _never_ blocks on add. i'm not 100% sure bitswap _never_ blocks. 


an add operation should be completely decoupled from providing to the network. maybe should be using notifications/events for this:

```
add() -> emits "added <key>" notification

provider listens to "added <key>" notification
```

**Comment by whyrusleeping** [comment]
@jbenet bitswap never blocks on add: https://github.com/ipfs/go-ipfs/blob/master/exchange/bitswap/workers.go#L124-L134

File: blockservice/blockservice.go [hunk 1]
```diff
@@ -67,7 +46,7 @@ func (s *BlockService) AddBlock(b *blocks.Block) (key.Key, error) {
 	if err != nil {
 		return k, err
 	}
-	if err := s.worker.HasBlock(b); err != nil {
+	if err := s.Exchange.HasBlock(context.TODO(), b); err != nil {
```
Comment (jbenet) [inline blockservice/blockservice.go]: thinking that this HasBlock is a passive activity, not an active one like the "AddBlock" call. i.e. user is adding, not wanting to write out. or send blocks to others. may be worth doing .HasBlock in a goroutine/process/worker.

context should be derived from some context passed to the blockservice initially, not the current request, i think.

**Comment by whyrusleeping** [comment]
@jbenet can has merge?

**Comment by jbenet** [comment]
needs rebase. assign me when done

**Comment by jbenet** [comment]
@whyrusleeping did it get out of date again?

**Comment by jbenet** [comment]
sorry

**Comment by whyrusleeping** [comment]
I can rebase again, this ones pretty easy compared to some of the others, lol

**Comment by whyrusleeping** [comment]
@jbenet rebased.

**Pull request merged**